src/components/chats/useUpdateChatTitle.tsx

```diff
@@ -0,0 +1,36 @@
+import { api } from '@/lib/api'
+import { debounce } from 'lodash'
+import { useEffect } from 'react'
+import { useErrorHandler } from '../global/errorHandlingHooks'
+
+export const useUpdateChatTitle = () => {
```

Comment:
Why not using `updatePost` directly? The good thing about updatePost is that updates the react-query cache directly; having a single source of truth.